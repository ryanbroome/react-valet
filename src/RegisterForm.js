import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import ThemeContext from "./ThemeContext";

const RegisterForm = () => {
  const history = useHistory();
  const { register } = useContext(ThemeContext);

  const INITIAL_STATE = {
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: "",
    phone: "",
    locationId: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // method to update parent state to trigger authentication and saving of token in context
    register(formData.username, formData.password, formData.firstName, formData.lastName, formData.email, formData.phone, formData.locationId);
    setFormData(INITIAL_STATE);
    // *
    history.push(`/`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>New User Form</h3>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
          <Input
            id={val}
            key={`Input-${val}`}
            type={val !== "email" ? "text" : "email"}
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}></Input>
        </FormGroup>
      ))}

      <Button color="primary">Register</Button>
    </Form>
  );
};

export default RegisterForm;
