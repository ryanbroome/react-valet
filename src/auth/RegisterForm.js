import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import UserContext from "./UserContext";

const RegisterForm = () => {
  const history = useHistory();
  const { register } = useContext(UserContext);
  //todo add a type="select" Input on form with each <option> being the different locations available to choose from</option>
  //todo update if formType object needed or write out each input if needed
  const INITIAL_STATE = {
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: "",
    phone: "",
    locationId: "",
  };

  let formType = {
    username: "text",
    password: "password",
    firstName: "text",
    lastName: "text",
    email: "email",
    phone: "phone",
    locationId: "text",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    register(formData.username, formData.password, formData.firstName, formData.lastName, formData.email, formData.phone, Number(formData.locationId));
    setFormData(INITIAL_STATE);
    history.push(`/transactions/active`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>New User Form</h3>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
          <Input
            id={val}
            key={`Input-${val}`}
            type={formType[val] ? formType[val] : "text"}
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}></Input>
        </FormGroup>
      ))}

      <Button color="primary">Register</Button>
    </Form>
  );
};

export default RegisterForm;
