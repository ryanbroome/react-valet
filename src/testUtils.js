import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
