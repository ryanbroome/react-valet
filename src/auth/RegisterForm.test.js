import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import RegisterForm from "./RegisterForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("Register Form: matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<RegisterForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Register Form: renders without crashing", function () {
  render(
    <MemoryRouter>
      <MockUserContext children={<RegisterForm />}></MockUserContext>
    </MemoryRouter>
  );
});
