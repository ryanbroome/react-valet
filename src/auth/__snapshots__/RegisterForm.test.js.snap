// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Form: matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <h3>
      New User Form
    </h3>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="username"
      />
      <input
        class="form-control"
        id="username"
        name="username"
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="password"
      />
      <input
        class="form-control"
        id="password"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="firstName"
      />
      <input
        class="form-control"
        id="firstName"
        name="firstName"
        placeholder="firstname"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="lastName"
      />
      <input
        class="form-control"
        id="lastName"
        name="lastName"
        placeholder="lastname"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="email"
      />
      <input
        class="form-control"
        id="email"
        name="email"
        placeholder="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="phone"
      />
      <input
        class="form-control"
        id="phone"
        name="phone"
        placeholder="phone"
        type="phone"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="locationId"
      />
      <input
        class="form-control"
        id="locationId"
        name="locationId"
        placeholder="locationid"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
    >
      Register
    </button>
  </form>
</DocumentFragment>
`;
